### PR TITLE
chore: add example of puppeteer

### PR DESCRIPTION
**Pull request by JustinBeckwith**
This was super fun!  Adds an example of using puppeteer to do automated testing of an OAuth2 workflow with Google Sign in. Addresses #225.

### CODE REVIEW

File: examples/puppeteer/oauth2-test.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
+// Copyright 2018, Google, Inc.
```
Comment (jinwoo) [inline examples/puppeteer/oauth2-test.js]: Google LLC. The copyright template was updated with Google LLC.

File: examples/puppeteer/oauth2-test.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+async function main() {
+  const oAuth2Client = await getAuthenticatedClient();
+  const url = 'https://www.googleapis.com/plus/v1/people?query=pizza';
+  const res = await oAuth2Client.request({ url });
+  console.log(res.data);
+}
+
+function getAuthenticatedClient() {
+  let browser = null;
+  return new Promise((resolve, reject) => {
```
Comment (jinwoo) [inline examples/puppeteer/oauth2-test.js]: Nit. `reject` is not needed.

File: examples/puppeteer/oauth2-test.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  await page.mainFrame().waitForSelector('#identifierNext');
+  console.log('clicking next button...');
+  await page.click('#identifierNext');
+  console.log('waiting for password field...');
+  await page
+    .mainFrame()
+    .waitForSelector('#password input[type="password"]', { visible: true });
+  console.log('typing password...');
+  await page.type('#password input[type="password"]', config.password, {
+    delay: 100
```
Comment (jinwoo) [inline examples/puppeteer/oauth2-test.js]: Why is a delay needed here?

File: examples/puppeteer/oauth2-test.js (hunk 1 above)
Comment (JustinBeckwith) [reply examples/puppeteer/oauth2-test.js]: fixed!

File: examples/puppeteer/oauth2-test.js (hunk 2 above)
Comment (JustinBeckwith) [reply examples/puppeteer/oauth2-test.js]: fixed!

File: examples/puppeteer/oauth2-test.js (hunk 3 above)
Comment (JustinBeckwith) [reply examples/puppeteer/oauth2-test.js]: UI automation jank.  Puppeteer is fairly new, and it was hard to get the timing of all this stuff to work right.

**Pull request merged**